packages/contracts/test/exchange/match_orders.ts

```diff
@@ -602,7 +602,7 @@ describe('matchOrders', () => {
             // Cancel left order
             await exchangeWrapper.cancelOrderAsync(signedOrderLeft, signedOrderLeft.makerAddress);
             // Match orders
-            return expectRevertReasonOrAlwaysFailingTransactionAsync(
+            return expectTransactionFailedAsync(
```

Comment:
On #796, @fabioberger asked:

> Does it still verify the balances?

(For context, I changed this from `matchOrderTester.matchOrdersAndVerifyBalancesAsync` because the new assertions are more strict about the type of Promises they accept).

No, but neither did the old version. If you look closely at the [implementation of `matchOrderTester.matchOrdersAndVerifyBalancesAsync`](https://github.com/0xProject/0x-monorepo/blob/5b64b3ea937326978b5742ec1b3692ebe5c41991/packages/contracts/test/utils/match_order_tester.ts#L111) it calls `await this._exchangeWrapper.matchOrdersAsync(signedOrderLeft, signedOrderRight, takerAddress);` prior to checking the balances. If `matchOrdersAsync` rejects, it does not check the balances. Maybe this was an oversight?

Reply:
@hysz was it an oversight?

Reply:
IIRC the idea was that if `matchOrdersAsync` throws, then the transaction reverted and balances need not be checked. Now that we enforce _revert reasons_ the tests are far more robust 👍.